include/exadg/structure/spatial_discretization/operators/nonlinear_operator.cpp

```diff
@@ -275,27 +282,83 @@ template<int dim, typename Number>
 void
 NonLinearOperator<dim, Number>::do_boundary_integral_continuous(
   IntegratorFace &                   integrator,
+  OperatorType const &               operator_type,
   dealii::types::boundary_id const & boundary_id) const
 {
   BoundaryType boundary_type = this->operator_data.bc->get_boundary_type(boundary_id);
 
   for(unsigned int q = 0; q < integrator.n_q_points; ++q)
   {
-    auto traction = calculate_neumann_value<dim, Number>(
-      q, integrator, boundary_type, boundary_id, this->operator_data.bc, this->time);
+    vector traction;
 
-    if(this->operator_data.pull_back_traction)
+    // integrate standard (stored) traction or exterior pressure on Robin boundaries
+    if(boundary_type == BoundaryType::Neumann or boundary_type == BoundaryType::NeumannCached or
+       boundary_type == BoundaryType::RobinSpringDashpotPressure)
     {
-      tensor F = get_F<dim, Number>(integrator.get_gradient(q));
-      vector N = integrator.get_normal_vector(q);
-      // da/dA * n = det F F^{-T} * N := n_star
-      // -> da/dA = n_star.norm()
-      vector n_star = determinant(F) * transpose(invert(F)) * N;
-      // t_0 = da/dA * t
-      traction *= n_star.norm();
+      if(operator_type == OperatorType::inhomogeneous or operator_type == OperatorType::full)
+      {
+        traction -= calculate_neumann_value<dim, Number>(
```

Comment:
I don't like that a minus sign is introduced here, and that the sign is also changed below in function `submit_value()`. 

(As a sidenote, the present change turns an assignment into an addition/subtraction, so why didn't you actually write `traction = - xyz;` to introduce a minus sign compared to the previous version? However, we should keep the previous version in my opinion. Robin is a new boundary type, that is optional.)